docs/upgrade.html

```diff
@@ -61,6 +61,25 @@ <h4><a id="upgrade_3_4_0" href="#upgrade_3_4_0">Upgrading to 3.4.0 from any vers
         </li>
     </ol>
 
+<h4><a id="upgrade_3_4_0" href="#upgrade_3_4_0">Upgrading a KRaft-based cluster to 3.4.0 from any version 3.0.x through 3.3.x</a></h4>
+
+    <p><b>If you are upgrading from a version prior to 3.4.0, please see the note below. Once you have changed the metadata.version to the latest version, it will not be possible to downgrade to a version prior to 3.4-IV0.</b></p>
+
+    <p><b>For a rolling upgrade:</b></p>
+
+    <ol>
+        <li>Upgrade the brokers one at a time: shut down the broker, update the code, and restart it. Once you have done so, the
+            brokers will be running the latest version and you can verify that the cluster's behavior and performance meets expectations.
+        </li>
+        <li>Once the cluster's behavior and performance has been verified, bump the metadata.version by running
+            <code>
+                ./bin/kafka-features.sh upgrade --metadata 3.4
+            </code>
+        </li>
+        <li>Note that the cluster metadata version cannot be downgraded to a pre-production 3.0.x, 3.1.x, 3.2.x, or 3.3.x version once it has been upgraded.
+            However, it is possible to downgrade to production versions such as 3.3-IV0, 3.3-IV1, etc.</li>
```

Comment:
hey @mumrah or @cmccabe or whoever would know best, can you double check this section and specifically this line? The KRaft upgrade section in 3.3 also had `3.3-IV0, 3.3-IV1` as examples but I wasn't sure if it would make sense to bump this to `3.4-IV0, 3.4-IV1`, or some combination of those?

Reply:
Pre-production versions are "3.0.x, 3.1.x, 3.2.x". That's before kraft was declared production ready.

Reply:
Gotcha, fixed the pre-prod version list. Still not sure what the `-IV0` suffixes are, but this make sense now?

Reply:
Yes, this part seems fine now. I had missed something else the first time round and left a comment